src/dmd/statement.d

```diff
@@ -449,6 +449,82 @@ extern (C++) abstract class Statement : ASTNode
     inout(CompoundDeclarationStatement) isCompoundDeclarationStatement() { return stmt == STMT.CompoundDeclaration ? cast(typeof(return))this : null; }
 }
 
+
+// TODO Make this function iterative if it turns out to be a bottleneck
+private Statement stripScopeStatment(Statement s)
+{
+    Statement result = s;
+
+    if (s && s.stmt == STMT.Scope)
+    {
+        result = s.isScopeStatement().statement;
+        result = stripScopeStatment(result);
+    }
+
+    return result;
+}
+/***********************************************************
+ * Determines if a CompundStatement with scopeStatements in contains assert(__ctfe)
+ */
+bool containsAssertCtfe(CompoundStatement stmt)
+{
+    bool result = false;
+
+    Statements *statements;
+    if (stmt !is null)
+    {
+        if (CompoundStatement cs = stmt.isCompoundStatement())
+        {
+            statements = cs.statements;
+        }
+    }
+
+    if (statements !is null)
+    {
+        foreach(s;(*statements))
```

Comment:
This loop should stop at the first non-null member; it should not look for any member.

Reply:
This loop is iterating over the in contracts.